src/coreclr/nativeaot/Runtime/gctoclreventsink.cpp

```diff
@@ -163,41 +161,43 @@ void GCToCLREventSink::FireGCAllocationTick_V4(uint64_t allocationAmount,
     LIMITED_METHOD_CONTRACT;
 
     void * typeId = RedhawkGCInterface::GetLastAllocEEType();
-    WCHAR * name = nullptr;
+    wchar_t * name = nullptr;
```

Comment:
This and all the definitions/implementations should use `WCHAR`, not `wchar_t`. (This is the problem that https://github.com/dotnet/runtime/pull/87465 was fixing.)

`wchar_t` is implementation-defined - so it ends up being UTF-16 on Windows (MSVC) and UTF-32 on non-Windows (clang, gcc). Our PAL defines `WCHAR` to be 2-byte character on all platforms.